sdk/lib/cmlib/cmcheck.c

```diff
@@ -318,7 +318,7 @@ CmpValidateValueListByCount(
                                          ValueListData,
                                          FixHive))
             {
-                DPRINT1("The value cell is NIL (at index %lu, list count %lu)\n",
+                DPRINT1("The value cell is NIL (at index %u, list count %u)\n",
```

Comment:
All those cmlib and cabman warnings can also be observed on our GCC8.4.0-**Linux**-builders, they are **not** limited to GCC13.2.0.
As we can see here: (this is a log from another PR gcc-i386-linux to demonstrate the state before Hermes PR):
https://github.com/reactos/reactos/actions/runs/11182169493/job/31087833648?pr=7425

They do not happen on RosBEWin2.2.2-**Windows** though.

I did mention this, because it allows us to use the github actions to verify the effectiveness of Hermes PR by simply using the github actions:
https://github.com/reactos/reactos/actions/runs/11093799616/job/30820527615?pr=7408

It shows that Hermes did properly manage to fix all warnings in cabman.
And it shows that some of those warnings in cmlib are fixed now, but it also shows that some are still present after the PR in cmlib:
```
2024-09-29T16:43:47.9889733Z /home/runner/work/reactos/reactos/src/sdk/lib/cmlib/cmcheck.c:1252:66: warning: format ‘%p’ expects argument of type ‘void *’, but argument 2 has type ‘HCELL_INDEX’ {aka ‘unsigned int’} [-Wformat=]
2024-09-29T16:43:47.9892287Z  1252 |                     DPRINT1("The key is corrupt (current cell 0x%p, parent cell 0x%p)\n",
2024-09-29T16:43:47.9893393Z       |                                                                 ~^
2024-09-29T16:43:47.9894296Z       |                                                                  |
2024-09-29T16:43:47.9895162Z       |                                                                  void *
2024-09-29T16:43:47.9896221Z       |                                                                 %d
2024-09-29T16:43:47.9897233Z  1253 |                             CurrentCell, ParentCell);
2024-09-29T16:43:47.9898115Z       |                             ~~~~~~~~~~~                           
2024-09-29T16:43:47.9898866Z       |                             |
2024-09-29T16:43:47.9899701Z       |                             HCELL_INDEX {aka unsigned int}
2024-09-29T16:43:47.9902210Z /home/runner/work/reactos/reactos/src/sdk/lib/cmlib/cmcheck.c:1252:84: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘HCELL_INDEX’ {aka ‘unsigned int’} [-Wformat=]
2024-09-29T16:43:47.9904728Z  1252 |                     DPRINT1("The key is corrupt (current cell 0x%p, parent cell 0x%p)\n",
2024-09-29T16:43:47.9936151Z       |                                                                                   ~^
2024-09-29T16:43:47.9937146Z       |                                                                                    |
2024-09-29T16:43:47.9938181Z       |                                                                                    void *
2024-09-29T16:43:47.9939157Z       |                                                                                   %d
2024-09-29T16:43:47.9940063Z  1253 |                             CurrentCell, ParentCell);
2024-09-29T16:43:47.9941095Z       |                                          ~~~~~~~~~~                                 
2024-09-29T16:43:47.9942158Z       |                                          |
2024-09-29T16:43:47.9943003Z       |                                          HCELL_INDEX {aka unsigned int}
2024-09-29T16:43:47.9945903Z /home/runner/work/reactos/reactos/src/sdk/lib/cmlib/cmcheck.c:1283:87: warning: format ‘%p’ expects argument of type ‘void *’, but argument 2 has type ‘HCELL_INDEX’ {aka ‘unsigned int’} [-Wformat=]
2024-09-29T16:43:47.9948541Z  1283 |                             DPRINT1("The lexicographical order is invalid (sibling 0x%p, current cell 0x%p)\n",
2024-09-29T16:43:47.9949896Z       |                                                                                      ~^
2024-09-29T16:43:47.9950795Z       |                                                                                       |
2024-09-29T16:43:47.9951694Z       |                                                                                       void *
2024-09-29T16:43:47.9952889Z       |                                                                                      %d
2024-09-29T16:43:47.9954210Z  1284 |                             CurrentCell, WorkState[StackDepth - CMP_PRIOR_STACK].Sibling);
2024-09-29T16:43:47.9955394Z       |                             ~~~~~~~~~~~                                                
2024-09-29T16:43:47.9956636Z       |                             |
2024-09-29T16:43:47.9957468Z       |                             HCELL_INDEX {aka unsigned int}
2024-09-29T16:43:47.9960155Z /home/runner/work/reactos/reactos/src/sdk/lib/cmlib/cmcheck.c:1283:106: warning: format ‘%p’ expects argument of type ‘void *’, but argument 3 has type ‘HCELL_INDEX’ {aka ‘unsigned int’} [-Wformat=]
2024-09-29T16:43:47.9962779Z  1283 |                             DPRINT1("The lexicographical order is invalid (sibling 0x%p, current cell 0x%p)\n",
2024-09-29T16:43:47.9964074Z       |                                                                                                         ~^
2024-09-29T16:43:47.9965127Z       |                                                                                                          |
2024-09-29T16:43:48.0010316Z       |                                                                                                          void *
2024-09-29T16:43:48.0011353Z       |                                                                                                         %d
2024-09-29T16:43:48.0012812Z  1284 |                             CurrentCell, WorkState[StackDepth - CMP_PRIOR_STACK].Sibling);
2024-09-29T16:43:48.0013954Z       |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                  
2024-09-29T16:43:48.0015003Z       |                                                                                 |
2024-09-29T16:43:48.0016196Z       |                                                                                 HCELL_INDEX {aka unsigned int}
2024-09-29T16:43:48.0018816Z /home/runner/work/reactos/reactos/src/sdk/lib/cmlib/cmcheck.c:1304:68: warning: format ‘%p’ expects argument of type ‘void *’, but argument 2 has type ‘HCELL_INDEX’ {aka ‘unsigned int’} [-Wformat=]
2024-09-29T16:43:48.0021549Z  1304 |             DPRINT1("Couldn't get the node of key (current cell 0x%p)\n", CurrentCell);
2024-09-29T16:43:48.0022703Z       |                                                                   ~^      ~~~~~~~~~~~
2024-09-29T16:43:48.0023604Z       |                                                                    |      |
2024-09-29T16:43:48.0024623Z       |                                                                    void * HCELL_INDEX {aka unsigned int}
2024-09-29T16:43:48.0025578Z       |                                                                   %d
2024-09-29T16:43:48.0026945Z [241/297] Building C object sdk/lib/cmlib/CMakeFiles/cmlibhost.dir/cminit.c.o
```

I will refine that PRs title, and I will approve it already. But I would also welcome if the remaining cmlib warnings could be fixed as well.

Reply:
There are 5 new warnings that I have to fix, though.